### PR TITLE
Improve ajax response when must renew needed (no more redirect)

### PR DESCRIPTION
**Pull request by galech**
PR now set to merge do dev/dj1.11.

Don't forget to forward merge or cherypick to `dev/dj2.2` and `master`

### CODE REVIEW

**Review by klausfmh — APPROVED** [review]

File: pwdtk/middlewares.py [hunk 1]
```diff
@@ -84,12 +85,25 @@ def process_exception(self, request, exception):
 
     def process_request(self, request):
 
-        if self.must_renew_password(request):
+        if self.must_renew_password(request, response=False):
```
Comment (klausfmh) [inline pwdtk/middlewares.py:88]: if response is normally a response object, then the default should be `None` and not `False`

if it is really a boolean, then perhaps better to chose a name like
`is_response` `has_response` or `response_flag`

Comment (galech) [reply pwdtk/middlewares.py]: refactored ;)

File: pwdtk/middlewares.py [hunk 2]
```diff
@@ -43,12 +43,7 @@ def must_renew_password(self, request):
         if request.method == 'GET':
             return False
 
-        if django.VERSION < (1, 10):
-            is_authenticated = request.user and request.user.is_authenticated()
-        else:
-            is_authenticated = request.user and request.user.is_authenticated
-
```
Comment (klausfmh) [inline pwdtk/middlewares.py:50]: I understand, why the if statement disappeared, byut are you sure now, that request.user is alwasy not None?

Comment (galech) [reply pwdtk/middlewares.py:50]: not sure but i think it will be a Anonymus users at least, never None

Comment (feenes) [reply pwdtk/middlewares.py:50]: any way to test this quickly (that you get an anonymous user object)?

**Review by asixela — APPROVED** [review]

File: pwdtk/helpers.py [hunk 3]
```diff
@@ -8,6 +8,15 @@
 from django.conf import settings
 
 
+def classname(obj):
+    cls = type(obj)
+    module = cls.__module__
+    name = cls.__qualname__
+    if module is not None and module != "__builtin__":
+        name = module + "." + name
```
Comment (klausfmh) [inline pwdtk/helpers.py:16]: `f"{module}.{name}"` is more efficient

**Review by klausfmh — APPROVED** [review]

**Pull request closed without merging**